doc/design/status_port.md

```diff
@@ -0,0 +1,453 @@
+associate the correct readers and writers a `ServiceDescription` is used.
+
+#### Frankenstein object corner cases
+
+How can we ensure in `StatusPortWriter::store()` that there is no `StatusPortReader`
+reading the chunk to which we want to write to? This could be for example a slow
+`StatusPortReader`. Is a reference counter needed? No, this is not the case. Instead
+the `StatusPortReader` checks if his view of on the world (aka the two chunks)
+has changed in-between.
+
+To illustrate this, let's assume that that there is one `StatusPortWriter` and one
+`StatusPortReader` in this scenario. Further assume that the type transferred via
+the `StatusPort` is
+
+```cpp
+T = cxx::string<36>
+```
+
+The following samples are transmitted by the `StatusPortWriter`
+
+1. "Time And Relative Dimension In Space"
+2. "Twin Ion Engines Fighter"
+3. "USS Enterprise"
+
+After the second transaction the `StatusPortReader` see's the latest transaction
+below and starts reading the `chunk[1]` aka `SECOND`.
+
+```text
+Latest transaction in shared memory managment segment
++----------------------------------+
+|       ActiveChunk::SECOND        |
+|       abaCounter: 1              |
++----------------------------------+
+```
+
+Now the the `StatusPortWriter` starts writing concurrently the third sample and
+overtakes the slower `StatusPortReader`.
+Due the latest transaction depicted below, `writePosition` is calculated to
+`chunks[1]` aka `SECOND`. Below you can find the location `^` at which the writer
+is currently writing and the reader is reading. As a result the strings are mixed
+up at this very moment in time, a so called Frankenstein object. While in this
+case, the reader might still read uncorrupted data if he finished before the
+writer, one can easily imagine that this can lead to very subtile, nasty bugs.
+To avoid this problem the `StatusPortReader` compares the initial transaction
+with the latest transaction and  re-calls the callable as long as they are not
+the same.
+
+```text
+Shared memory payload segment
++----------------------------------------------------+
+|        SharedChunk chunks[2]                       |
+|        +----------------------------------------+  |
+| FIRST  | "Twin Ion Engines Fighter"             |  |
+|        +----------------------------------------+  |
+| SECOND | "USS Enterprlative Dimension In Space" |  |
+|        +----------------------------------------+  |
+|                      ^                 ^           |
+|                      Writer            Slow reader |
++----------------------------------------------------+
+
+Latest transaction in shared memory managment segment
++----------------------------------+
+|       ActiveChunk::FIRST         |
+|       abaCounter: 2              |
++----------------------------------+
+```
+
+> ***
+> Special care has to be taken, when writing the `callable`!
+> ***
+>
+> It must be ensured that the `callable` does not crash or ends up in an endless
```

Comment:
How does one verify this in a callable. One suggestion would be the following:
1. Create an integration tests which every callable has to pass.
2. This integration test has two threads, one low prio thread in which the callable is called, one high prio thread which randomly changes the data with random content. This test is not allowed to segfault.

Reply:
Brilliant idea! Probably such an integration test would need to run for several days to be sure to have statistically meaningful data to argue for minor probability of problems, right?